### ISSUE TITLE
Review GIBCT Caution Flag designs for accessibility

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#4919]
## Description
The UX team is working on implementing new Caution Flags into the GIBCT search results. DEPO is working on updating card patterns, and one of those deals with redesigning how warning alerts look in the comparison tool. For our short term solution, we’re looking to implement the design for the caution alert only without changing too much of the design for the rest of the card.

The team has two concerns:

* Our concern for the standard institutes is where we’re placing the caution alert relative to the location, and how that could affect users with screen readers. We have an example showing the alert beneath the name of the school, and one where it’s beneath the location. Seeing the name and then the location may be preferable for users with screen readers so they can know right away which school the alert is for.
* The other concern is for the all caps school names themselves. Right now, we can’t really do much about how the data is coming in, but since the names are linked, ideally, they’d be underlined. Underlined all caps bold text is a design concern, though, so we wanted to get your thoughts on that.

Please DM Trevor for the share link.

/cc @CrystabelReiter

**Comment by 1Copenut** [comment]
I'll leave my thoughts as two separate comments. Each comment will address one of the team's points of concern.

**Point 1: Location of alert in relation to the school location** (Screens 4-7)
I feel the location should be connected to the number of GI Bill students / city the school is located. Otherwise that data point feels de-coupled from its label.

I agree we want the screen reader to announce the school, the location, then the caution alerts in that order. I think we can accomplish this **and** keep the location with its data point using the [aria-describedby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-describedby_attribute#Example_2_A_Close_Button) attribute:

```html
<h3>
  <a
    aria-describedby="location-1"
    href="LINK_TO_SCHOOL"
  >Cortiva Institute</a>
<h3>

<!-- Alerts would go here -->

<!-- IDs must be unique or this will throw an accessibility error -->
<p id="location-1">
  Cromwell, <abbr title="Connecticut">CT</abbr>
</p>
```

* I added some optional markup to hopefully announce the full state name instead of the abbreviation. That's a nice to have.
* We will want to test this with screen reader users, ideally with different visual acuities. I want to confirm that announcing the location ahead of alert isn't confusing in situations where users have sight and use a screen reader.

**Comment by 1Copenut** [comment]
**Point 2: Readability of all caps linked text**
I did some digging, and I think we have some grounds to keep the text as presented in the InVision comps:

* [How to Meet WCAG 2.0: Use of Color](https://www.w3.org/WAI/WCAG20/quickref/20160105/#visual-audio-contrast-without-color)
* [G182: Ensuring that additional visual cues are available when text color differences are used to convey information](https://www.w3.org/WAI/GL/2016/WD-WCAG20-TECHS-20160105/G182)

The second link specifically calls out a few things that are of interest. Bold text mine:

> To use this technique, an author incorporates a visual cue in addition to color for each place where color alone is used to convey information. Visual cues can take many forms including **changes to the font style,** the addition of underlines, bold, or italics, or changes to the font size.

Our linked text looks to be using Bitter, and is a larger size than the text around it. It's not my ideal, but I agree that underlining all caps text is going to increase readability issues. Do we underline links on hover anywhere else on the site? It might be a nice medium, to underline text on hover, and put our yellow focus halo around it on keyboard focus.

Long-term it'd be better if the data stream can provide us text not in all caps, but I know that's out of the team's hands, and would be a longer time horizon.

**Comment by 1Copenut** [comment]
Great questions! Let me know if you want to chat and we can get together in a video call or Slack up.

**Comment by amyknox** [comment]
This was great info, @1Copenut, and very helpful to move us forward.  What you've outlined seems pretty straight-forward. We'll let you know if we have any questions. Thanks!

**Comment by hughes-dustin** [comment]
@1Copenut 
I'm finding that using `aria-describedby` is being ignored by OSX's Voice Over screen reader in both Chrome and FireFox. 

If we want the name and location to be included when focus is on the anchor tag, then we could use `aria-label`.
```                  
  <a href={linkTo.pathname} 
        aria-label={`${name} ${locationInfo(city, state, country)}`}>```

This definitely works with screen readers.

**Comment by 1Copenut** [comment]
@hughes-dustin Good call. I can see a couple of options here:

1. Moving the `aria-describedby` attribute up to the H3 to get VoiceOver to recognize it.
2. Using the `.vads-u-visibility--screen-reader ` class to wrap a span inside the A tag for screen readers. I'm starting to move away from `aria-label` because it [doesn't translate well](https://adrianroselli.com/2019/11/aria-label-does-not-translate.html).

LMK if you'd like to discuss tomorrow.
